### PR TITLE
feat: Set file reminders

### PR DESCRIPTION
**Pull request by Pytal**
- For https://github.com/nextcloud/server/issues/39181

- Base branch (merge first) https://github.com/nextcloud/server/pull/39651

## Summary

- [x] Register action
- [x] Set reminder
- [x] Clear reminder
- [x] Service for OCS API
- [x] DateTime utils

## TODO

- [x] Fix tests
- [x] Adjust "Later today"

## Screenshots

| | Screenshot |
--- | ---
**File action** | image
**Set reminder** | image
**Notification** | image

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/files_reminders/src/components/SetReminderActions.vue [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+-->
+
+<template>
+	<NcActions class="actions-secondary-vue"
+		:open.sync="open">
+		<NcActionButton @click="$emit('back')">
+			<template #icon>
+				<NcIconSvgWrapper :svg="arrowLeftSvg" />
```
Comment (susnux) [inline apps/files_reminders/src/components/SetReminderActions.vue:28]: I would use the material-design-icon components instead, and only use svg + wrapper for dynamic icons, as this adds additional runtime by validating / sanitizing the SVG.

File: apps/files_reminders/src/components/SetReminderActions.vue [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+		},
+	},
+})
+</script>
+
+<style lang="scss" scoped>
+.actions-secondary-vue {
+	display: block !important;
+    float: right !important;
+    padding: 5px 0 0 4px !important;
```
Comment (susnux) [inline apps/files_reminders/src/components/SetReminderActions.vue:222]:
```suggestion
	float: right !important;
	padding: 5px 0 0 4px !important;
```

**Review by susnux — APPROVED** [review]
Looks good, just a minor indention issue. And maybe use icon components instead of wrapped svgs (otherwiese on every mount `sanitizeSVG` is executed).

**Review by nfebe — APPROVED** [review]
Looks good!

File: apps/files/js/fileactions.js [hunk 3]
```diff
@@ -710,6 +710,22 @@
...
+					displayName: function(_context) {
+						return t('files', 'Set reminder');
+					},
+					mime: 'all',
+					order: -24,
+					icon: function(_filename, _context) {
+						return OC.imagePath('files_reminders', 'alarm.svg')
+					},
+					permissions: OC.PERMISSION_READ,
+					actionHandler: function(_filename, _context) {},
```
Comment (skjnldsv) [inline apps/files/js/fileactions.js:725]: Hey @Pytal, why is there no action handler here?

Comment (skjnldsv) [reply apps/files/js/fileactions.js:725]: Wow, I get it now, why not registering the action from `apps/files_reminders/src/main.ts` ? ^^
`OCA.Files.App.fileList.fileActions.registerAction()`

Comment (skjnldsv) [reply apps/files/js/fileactions.js:725]: This is legacy :)

the f2v way is
```js
import { registerFileAction } from `@nextcloud/files`
registerFileAction({...})
```

Comment (Pytal) [reply apps/files/js/fileactions.js:725]: Indeed, by f2v I meant when we implement this action with the future nested actions API not the implementation added in this PR ;)

Comment (skjnldsv) [reply apps/files/js/fileactions.js:725]: AH yes! :)

**Pull request merged**